Examples/Sources/MacroExamples/Implementation/Member/MetaEnumMacro.swift

```diff
@@ -22,18 +22,22 @@ public struct MetaEnumMacro {
   let parentParamName: TokenSyntax
 
   init(node: AttributeSyntax, declaration: some DeclGroupSyntax, context: some MacroExpansionContext) throws {
-    guard let enumDecl = declaration.as(EnumDeclSyntax.self) else {
+    guard var enumDecl = declaration.as(EnumDeclSyntax.self) else {
       throw DiagnosticsError(diagnostics: [
         CaseMacroDiagnostic.notAnEnum(declaration).diagnose(at: Syntax(node))
       ])
     }
 
-    parentTypeName = enumDecl.name.with(\.trailingTrivia, [])
+    enumDecl.name.trailingTrivia = []
+
+    parentTypeName = enumDecl.name
 
     access = enumDecl.modifiers.first(where: \.isNeededAccessLevelModifier)
 
     childCases = enumDecl.caseElements.map { parentCase in
-      parentCase.with(\.parameterClause, nil)
+      var parentCase = parentCase
+      parentCase.parameterClause = nil
+      return parentCase
```

Comment:
Similarly here where we are creating a copy with only one property modified, I think `with` is actually nicer. 
I just think that `with` doesn’t scale if you are modifying nested properties or multiple.